hie-bios/src/HIE/Bios/Load.hs

```diff
@@ -35,7 +35,7 @@ loadFileWithMessage :: GhcMonad m
 loadFileWithMessage msg file = do
   dir <- liftIO $ getCurrentDirectory
   pprTraceM "loadFile:2" (text dir)
-  withDynFlags (setWarnTypedHoles . setDeferTypeErrors . setNoWaringFlags) $ do
+  withDynFlags (setWarnTypedHoles . setNoWaringFlags) $ do
```

Comment:
The intention is to get as many warnings/errors as we can, even in the presence of a partially broken source file.

Reply:
For some reason the warnings were not being displayed, possibly because `-Wall` was not enabled. Needs some more investigation.

Reply:
`Opt_WarnDeferredTypeErrors` and `Opt_WarnDeferredOutOfScopeVariables` are the flags that control this.

I think if we have this, we should report the warning as an error if the `WarnReason` is one of these.

Reply:
I think the `WarnDeferred*` flags are implied when they are parsed from the command line options, but here we are manually modifying the `DynFlags`, so those have to be explicitly set.

So it would be a good idea to explictly set all of `Opt_DeferTypeErrors`, `Opt_DeferOutOfScopeVariables`, `Opt_WarnDeferredTypeErrors` and `Opt_WarnDeferredOutOfScopeVariables`.